### DISCUSSION TITLE
Why does GetLifecycleConfig return an error when there are no lifecycle rules?

### DISCUSSION THREAD

**Discussion by Bog999** [minio/minio#12921] [Q&A] [answered]
The GetLifecycleConfig function that maps to the `ilm ls <bucket-name>` command returns an error (NoSuchLifecycleConfiguration) when the list of rules is empty. Shouldn't it return an empty list?

**Comment by vadmeste** [comment, marked as answer]
> The GetLifecycleConfig function that maps to the `ilm ls <bucket-name>` command returns an error (NoSuchLifecycleConfiguration) when the list of rules is empty. Shouldn't it return an empty list?

MinIO follows S3 specification here.

```
mc: <DEBUG> GET /?lifecycle= HTTP/1.1
...
...

mc: <DEBUG> HTTP/1.1 404 Not Found
...
...

<Error><Code>NoSuchLifecycleConfiguration</Code><Message>The lifecycle configuration does not exist</Message><BucketName>testbucket</BucketName><RequestId>HDX8DQJ4P5WNZGCT</RequestId><HostId>VzXVbOHVDI3GPbujA2ftydQ0UjmpY4Fkupm//RMFvD4a+AFBbU+sq4T0BD7YphqyRggAIS6LkZ0=</HostId></Error>
```

**Reply by Bog999** [reply]
Thanks!

**Comment by Bog999** [comment]
Is there an easy way to get the HTTP Error code from the minio client when I run the `mc --json ilm ls <bucket-name>` command? I know that the --debug option shows it but it requires parsing.

**Reply by harshavardhana** [reply]
> Is there an easy way to get the HTTP Error code from the minio client when I run the `mc --json ilm ls <bucket-name>` command? I know that the --debug option shows it but it requires parsing.

If you want to see that you can use SDKs directly, we do not expose lower HTTP status codes.
